march_hardware/test/TestEncoder.cpp

```diff
@@ -29,54 +41,67 @@ class TestEncoderParameterizedValidTarget : public TestEncoder,
 {
 };
 
-TEST_F(TestEncoder, CorrectLimits)
+TEST_F(TestEncoder, CorrectLowerHardLimits)
 {
   ASSERT_EQ(this->encoder.getLowerHardLimitIU(), this->lower_limit);
+}
+
+TEST_F(TestEncoder, CorrectUpperHardLimit)
+{
   ASSERT_EQ(this->encoder.getUpperHardLimitIU(), this->upper_limit);
 }
 
-TEST_F(TestEncoder, CorrectSoftLimits)
+TEST_F(TestEncoder, CorrectLowerSoftLimits)
 {
-  ASSERT_EQ(this->encoder.getLowerSoftLimitIU(), 3144);
-  ASSERT_EQ(this->encoder.getUpperSoftLimitIU(), 44650);
+  ASSERT_EQ(this->encoder.getLowerSoftLimitIU(), this->lower_soft_limit);
 }
 
-TEST(TestEncoderConstructor, ResolutionBelowRange)
+TEST_F(TestEncoder, CorrectUpperSoftLimits)
 {
-  ASSERT_THROW(march::Encoder(0, 0, 0, 0, 0, 0, 0), march::error::HardwareException);
+  ASSERT_EQ(this->encoder.getUpperSoftLimitIU(), this->upper_soft_limit);
 }
 
-TEST(TestEncoderConstructor, ResolutionAboveRange)
+TEST_F(TestEncoder, ResolutionBelowRange)
 {
-  ASSERT_THROW(march::Encoder(50, 0, 0, 0, 0, 0, 0), march::error::HardwareException);
+  ASSERT_THROW(march::Encoder(0, this->lower_limit, this->upper_limit, this->lower_limit_rad, this->upper_limit_rad,
+                              this->lower_soft_limit_rad, this->upper_soft_limit_rad),
+               march::error::HardwareException);
 }
 
-TEST(TestEncoderConstructor, DifferentIuPerRad)
+TEST_F(TestEncoder, ResolutionAboveRange)
 {
-  ASSERT_THROW(march::Encoder(12, 2, 1, 1.0, 2.0, 1.0, 2.0), march::error::HardwareException);
+  ASSERT_THROW(march::Encoder(50, this->lower_limit, this->upper_limit, this->lower_limit_rad, this->upper_limit_rad,
+                              this->lower_soft_limit_rad, this->upper_soft_limit_rad),
+               march::error::HardwareException);
 }
 
-TEST(TestEncoderConstructor, InvalidSoftLimits)
+TEST_F(TestEncoder, LowerSoftLimitAboveUpperSoftLimit)
 {
-  ASSERT_THROW(march::Encoder(17, 2053, 45617, -0.34906585, 1.745329252, 0.4, 0.1), march::error::HardwareException);
+  ASSERT_THROW(march::Encoder(this->resolution, this->lower_limit, this->upper_limit, this->lower_limit_rad,
+                              this->upper_limit_rad, 0.4, 0.1),
+               march::error::HardwareException);
 }
 
-TEST(TestEncoderConstructor, InvalidLowerSoftLimit)
+TEST_F(TestEncoder, LowerSoftLimitLowerThanLowerHardLimit)
 {
-  ASSERT_THROW(march::Encoder(17, 2053, 45617, -0.34906585, 1.745329252, -0.4, 1), march::error::HardwareException);
+  ASSERT_THROW(march::Encoder(this->resolution, this->lower_limit, this->upper_limit, this->lower_limit_rad,
+                              this->upper_limit_rad, -0.4, 1),
```

Comment:
As I said earlier:
> Unfortunately, I was not able to find a nice way to implement this, since the INSTANTIATE_TEST_CASE_P cannot access the TestEncoder static or member variables.

This is not possible and I was not able to find a way to do this :cry:

Reply:
Sorry, missed that comment